### ISSUE TITLE
MPD2 Widget spams .xsession_errors when tags are missing

### ISSUE THREAD

**Issue by MagusLucius** [qtile/qtile#1304]
The title says it all.  If the tags on a file are missing or malformed, the widget displays an unattractive and unhelpful "Error" in the bar and nothing else, and spams .xsession_errors with messages from an exception that is raised every time the widget updates.

Would it be possible, instead, to have the widget handle the exception gracefully by outputting "Unknown" for the missing tags and display the output as normal?

Edit: I fixed the issue locally in the following way:
global:
```py
import re
```
at `__init__`
```py
tags = ['name', 'artist', 'album', 'albumartist', 'comment', 'composer',\
  'date', 'disc', 'genre', 'performer', 'title', 'track', 'time', 'file', 'position',\
  'mtime', 'mdate']
tags_re = re.compile('|'.join(tags))
...
self.watchwords = re.findall(tags_re, self.status_format)
```

in formatter:
```py
# add dummy value for missing tags.
for w in self.watchwords:
  if w not in status:
    status[w] = "Unknown"
```

I'd hate to step on any toes, and I've never contributed to anything, so I thought I'd post this here.

**Comment by ramnes** [comment]
Related PR: #1306

**Comment by m-col** [comment]
Was this closed by #1306?

**Comment by ramnes** [comment]
Let's assume so. Feel free to comment if needed and I'll reopen. :)
